src/jit/assertionprop.cpp

```diff
@@ -5153,18 +5203,11 @@ void Compiler::optAssertionPropMain()
                                                        // and thus we must morph, set order, re-link
             for (GenTree* tree = stmt->GetTreeList(); tree != nullptr; tree = tree->gtNext)
             {
-                if (tree->OperIs(GT_JTRUE))
-                {
-                    // A GT_TRUE is always the last node in a tree, so we can break here
-                    assert((tree->gtNext == nullptr) && (stmt->GetNextStmt() == nullptr));
-                    break;
```

Comment:
Hmm, can't remember exactly why I added this early exit for `JTRUE` nodes. Most likely for consistency with the assertion gen loop that has special handling for `JTRUE`. Here the only potentially unwanted side effect of not exiting seems to be setting the wrong bit in `assertions`. But since it's the last node of the block it doesn't matter...